functions/src/SQR.functions.js

```diff
@@ -0,0 +1,234 @@
+
+const functions = require('firebase-functions');
+const admin = require("firebase-admin");
+
+const bucket = admin.storage().bucket();
+const db = admin.database();
+const helpers = require('./../helpers');
+
+
+
+const SQR_Helper = require('./SQR_Helper');
+/////////////////////////////////////////////////
+//          OnNewAllotment (DB create and update Trigger)
+//      1. Mark the files in the database --> { status: "Given" }
+//              Function --> UpdateFilesOnNewAllotment
+//
+//      2. Send an email to the devotee to notify them of the new allotments
+//              Function --> sendEmailOnFileAllotment
+/////////////////////////////////////////////////
+
+exports.updateFilesOnNewAllotment = functions.database.ref('/sqr/allotments/{allotment_id}')
+    .onCreate((snapshot, context) => {
+        return SQR_Helper.UpdateFilesOnNewAllotment(
+                snapshot, 
+                db,
+                helpers.updateFile
+            );
+});
+
+exports.sendEmailOnFileAllotment = functions.database.ref('/sqr/allotments/{allotment_id}')
+    .onUpdate((change, context) => {
+        const old = change.before.val();
+        const _new = change.after.val();        
+        let coordinatorConfig = functions.config().audioseva.coordinator;
+        
+        db.ref('/sqr/allotments').once('value')
+        .then(snapshot => {
+            const allotments = snapshot.val();
+            let alloted_num = 0, alloted_before = false;
+            for(let key in allotments) {
```

Comment:
There should be space after `for`. The same for `if` etc. Make consistent everywhere please.

Reply:
And more importantly, are you really iterating all the allotments? Consider querying in order to get previous allotments and submissions.

Reply:
AFAIK it's possible. However consider creating an index in the db rules file in the root.